bool up
bool left
bool down
bool right

bool triangle
bool square
bool cross
bool circle

bool select
bool start
bool ps
bool trackpad

bool l1
bool l2
bool l3
bool r1
bool r2
bool r3

float32 analog_l2
float32 analog_r2

float32 analog_x1
float32 analog_y1
float32 analog_x2
float32 analog_y2